src/librustc/middle/const_eval.rs

```diff
@@ -877,12 +962,57 @@ pub fn eval_const_expr_partial<'tcx>(tcx: &ty::ctxt<'tcx>,
       ast::ExprBlock(ref block) => {
         match block.expr {
             Some(ref expr) => try!(eval_const_expr_partial(tcx, &**expr, ety)),
-            None => const_int(0)
+            None => const_int(0) // huh???
```

Comment:
added in 24ece07cecf3df59ce7b9ed3b317e3ffd718f756

I believe it should be `unreachable!()` or a `sess.span_bug` but I might be wrong

cc @Kimundi